Fakebook.Profile/Fakebook.Profile.UnitTests/APITests/ProfilePictureControllerTests.cs

```diff
@@ -99,8 +77,19 @@ public async Task UploadInvalidImageExtension(string extension)
                 ControllerContext = controllerContext
             };
 
-            // act and assert
-            Assert.IsType<BadRequestResult>(await controller.UploadProfilePicture());
+            if (successExpected)
+            {
+                // act
+                var result = await controller.UploadProfilePicture();
+
+                // assert
+                Assert.IsNotType<BadRequestResult>(await controller.UploadProfilePicture());
```

Comment:
calling method twice?

the act doesn't need to be conditional, just the assert.
prefer assert.isassignableto for the sake of liskov substitution principle
and isnottype is pretty broad, instead decide what type you do want to return on success

Reply:
removed double method call, only put the condition on assert and used isAssignable